commands/commands.go

```diff
@@ -88,23 +90,38 @@ func confirmInput(msg string) bool {
 	return false
 }
 
-func newProvider(store libmachine.Store) (*libmachine.Provider, error) {
-	return libmachine.New(store)
-}
-
 func getMachineDir(rootPath string) string {
 	return filepath.Join(rootPath, "machines")
 }
 
-func getDefaultStore(rootPath, caCertPath, privateKeyPath string) (libmachine.Store, error) {
-	return libmachine.NewFilestore(
-		rootPath,
-		caCertPath,
-		privateKeyPath,
-	), nil
+func getStore(c *cli.Context) persist.Store {
+	certInfo := getCertPathInfo(c)
+	return &persist.Filestore{
+		Path:           c.GlobalString("storage-path"),
+		CaCertPath:     certInfo.CaCertPath,
+		PrivateKeyPath: certInfo.CaKeyPath,
+	}
 }
 
-func setupCertificates(caCertPath, caKeyPath, clientCertPath, clientKeyPath string) error {
+func getFirstArgHost(c *cli.Context) *host.Host {
+	store := getStore(c)
+	hostName := c.Args().First()
+	h, err := store.Get(hostName)
+	if err != nil {
+		// I guess I feel OK with bailing here since if we can't get
+		// the host reliably we're definitely not going to be able to
+		// do anything else interesting, but also this premature exit
+		// feels wrong to me.  Let's revisit it later.
+		log.Fatalf("Error trying to get host %q: %s", hostName, err)
+	}
+	return h
+}
+
+func setupCertificates(certPathInfo cert.CertPathInfo) error {
```

Comment:
Calls to this function involved separately passing all of the parameters of this object, so I went ahead and cleaned that up.